src/dune/action.ml

```diff
@@ -272,7 +272,7 @@ type is_useful =
   | Clearly_not
   | Maybe
 
-let is_useful_to memoize =
+let is_useful_to distribute memoize =
```

Comment:
What is the meaning of the two arguments? Do all four variants make sense? I can't wrap my head around what's happening here, and how the various choices are made. Please add some comments.

Reply:
I have documented the public methods with the meaning of `useful_to_{sandbox,distribute,memoize}`. This function is just a helper to factor the traversal of actions, the only three possible invocations are juste below. Right now sandbox > distribute > memoize, but this is just by chance I think, some case that are not globally orderable might arise.